src/azul/terra.py

```diff
@@ -478,20 +479,30 @@ def _check_response(self,
         else:
             raise RequirementError('Unexpected response from TDR API', response.status)
 
+    page_size: ClassVar[int] = 100
+
     def snapshot_names_by_id(self) -> Dict[str, str]:
         """
         List the TDR snapshots accessible to the current credentials.
         """
-        limit = 100
         endpoint = self._repository_endpoint('snapshots')
-        response = self._request('GET', endpoint, fields=dict(limit=str(limit)))
-        response = self._check_response(endpoint, response)
-        # FIXME: Implement paging
-        #        https://github.com/DataBiosphere/azul/issues/3093
-        require(response['total'] <= limit, 'Too many snapshots', response['total'])
+        snapshots = []
+        while True:
+            response = self._request('GET', endpoint, fields={
+                'offset': len(snapshots),
+                'limit': self.page_size
+            })
+            response = self._check_response(endpoint, response)
+            new_snapshots = response['items']
+            if new_snapshots:
+                snapshots += new_snapshots
+            else:
+                total = response['filteredTotal']
```

Comment:
This would have failed agains `prod` TDR. What saves you is the fact that they just promoted that change to prod earlier today.

Either way, this creates and inconsistent state where some of the code attempts to handle the absence of `filteredTotal` and is preceded by a FIXME referencing https://github.com/DataBiosphere/azul/issues/3195 while this just assumes `filtererdTotal` exists.

Since it doesn't make sense to have this section gracefully handle the absence of `filteredTotal`, a case now not observable in the wild, we need to resolve that FIMXE in this PR as the first commit in the branch.